BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -706,47 +706,46 @@ public override void OnStart(StartState state)
                 animStates = GUIUtils.SetUpAnimation(flightAnimationName, part);
             }
 
-            IEnumerator<PartModule> partModules = part.Modules.GetEnumerator();
-            while (partModules.MoveNext())
+            warheadType = WarheadTypes.Kinetic; // Default to Kinetic if no appropriate modules are found.
+            foreach (var partModule in part.Modules)
             {
-                if (partModules.Current == null) continue;
-                if (partModules.Current.moduleName == "BDExplosivePart")
+                if (partModule == null) continue;
+                switch (partModule.moduleName)
                 {
-                    ((BDExplosivePart)partModules.Current).ParseWarheadType();
-                    if (((BDExplosivePart)partModules.Current).warheadReportingName == "Continuous Rod")
-                    {
-                        warheadType = WarheadTypes.ContinuousRod;
-                    }
-                    else
-                    {
-                        if (((BDExplosivePart)partModules.Current).tntMass > 0)
+                    case "BDExplosivePart":
+                        ((BDExplosivePart)partModule).ParseWarheadType();
+                        if (((BDExplosivePart)partModule).warheadReportingName == "Continuous Rod")
+                            warheadType = WarheadTypes.ContinuousRod;
+                        else
                             warheadType = WarheadTypes.Standard;
+                        break;
+                    case "ClusterBomb":
+                        clusterbomb = ((ClusterBomb)partModule).submunitions.Count;
+                        break;
+                    case "MultiMissileLauncher":
+                        if (weaponClass == WeaponClasses.Bomb)
+                            clusterbomb *= (int)((MultiMissileLauncher)partModule).salvoSize;
```

Comment:
If the MML module is before the ClusterBomb module in the config, this line will have no effect.
Presumably, cluster bombs need to also have a valid warhead type in their configs.

Reply:
I'm not sure why that was added, to be honest...
under normal circumstances, you shouldn't have a Clusterbomb and MultiMissileLauncher module on the same part. You would have a MML launching cluster ordinance, in which case you'd want the total number of clusterlets the launcher would put out counted, but that would be a if((MultiMissileLauncher)partModule).missileSpawner.missilePart.FindModuleImplementing<MissileLauncher>().clusterbomb test.

Clusterbombs would need a BDExplosiveModule, yes. Which should be getting grabbed by the switch statement here